### PR TITLE
Cleaned up some logic around messages and autoscroll

### PR DESCRIPTION
**Pull request by filbabic**
#2843 #2842 #2808 

This is a pick from #2844 so we can close both of them once merged.

### 🧪 Testing

When we're around up to 2 messages away from the bottom receiving new messages from other people should scroll to the bottom.

When we're scrolled away more than that, the visible scroll position should persist.

#### General
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

**Review by MarinTolic — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] ~~UI Changes correct (before & after images)~~
- [x] Bugs validated (bugfixes)
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [ ] ~~All code we touched has new or updated KDocs~~ (not necessary)

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/Messages.kt [hunk 1]
```diff
@@ -105,17 +107,20 @@ public fun Messages(
             }
         }
 
-        val firstVisibleItemIndex = currentListState.firstVisibleItemIndex
-
         when {
+            newMessageState == Other && firstVisibleItemIndex < 3 -> {
```
Comment (bychkovdmitry) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/Messages.kt:111]: The `animateScrollToItem(0)` is called several times. Could you check with if something can be done with that? For example, we can check for `!currentListState.isScrollInProgress`.
```
newMessageState == Other && firstVisibleItemIndex < 3 !currentListState.isScrollInProgress ->
```

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/Messages.kt [hunk 2]
```diff
@@ -105,17 +107,20 @@ public fun Messages(
...
+                coroutineScope.launch {
+                    currentListState.animateScrollToItem(0)
+                }
+            }
             newMessageState == MyOwn -> coroutineScope.launch {
```
Comment (bychkovdmitry) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/Messages.kt:116]: Same here:
```
newMessageState == MyOwn &&  !currentListState.isScrollInProgress -> coroutineScope.launch {
```

**Review by bychkovdmitry — CHANGES_REQUESTED** [review]

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/Messages.kt (hunk 1 above)
Comment (filbabic) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/Messages.kt:111]: That would be great - thanks for reminding me of this! :]

We can certainly test this, see how it behaves with a high load of messages.

**Review by bychkovdmitry — APPROVED** [review]
Looks good!

**Pull request merged**